crates/fsrt/src/main.rs

```diff
@@ -406,16 +325,18 @@ fn scan_directory(dir: PathBuf, function: Option<&str>, opts: &Args) -> Result<(
     if run_permission_checker && !perm_interp.permissions.is_empty() {
         reporter.add_vulnerabilities([PermissionVuln::new(perm_interp.permissions)]);
     }
-    let report = serde_json::to_string(&reporter.into_report()).into_diagnostic()?;
```

Comment:
You can probably return `Report` instead of the `Vec` of `Vulnerability`s here. That way, you can move the reporting logic out of the function, so that we don't unnecessarily print the report to stdout during testing. You can add a method like `pub fn into_vulnerabilities(self) -> Vec<Vulnerability>` to `Report`, so that the testing framework can take just the vulns.

Reply:
Return the reporter here right? Report I think is just a string ...

Reply:
By report, I mean the `Report` type, which is the result of `reporter.into_report()`.

Reply:
Makes sense. Thank you 😃

Reply:
I was looking at the report variable. But, now it makes sense 🙂

Reply:
Please check this, I think I may have fixed it. Thanks :)